appclient/src/main/java/org/jboss/as/appclient/service/ApplicationClientDeploymentService.java

```diff
@@ -32,68 +32,66 @@
-import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
 
-import org.jboss.as.controller.ModelController;
-import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.LocalModelControllerClient;
+import org.jboss.as.controller.ModelControllerClientFactory;
 import org.jboss.as.controller.client.Operation;
 import org.jboss.as.controller.client.OperationBuilder;
 import org.jboss.as.controller.operations.common.Util;
 import org.jboss.as.server.deployment.DeploymentAddHandler;
 import org.jboss.as.server.deployment.DeploymentDeployHandler;
 import org.jboss.dmr.ModelNode;
-import org.jboss.msc.service.Service;
-import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.Service;
 import org.jboss.msc.service.StartContext;
-import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
-import org.jboss.msc.value.InjectedValue;
 
 /**
  * Service responsible for deploying the application client that was specified on the command line
  *
  * @author Stuart Douglas
  */
-public class ApplicationClientDeploymentService implements Service<ApplicationClientDeploymentService> {
+public class ApplicationClientDeploymentService implements Service {
 
 
-    public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append("appClientDeploymentService");
-
     private final File path;
-    private ModelControllerClient controllerClient;
-    private final InjectedValue<ModelController> controllerValue = new InjectedValue<ModelController>();
+    private final Consumer<ApplicationClientDeploymentService> consumer;
+    private final Supplier<ModelControllerClientFactory> clientFactorySupplier;
+    private final Supplier<ExecutorService> executorServiceSupplier;
     private final CountDownLatch deploymentCompleteLatch = new CountDownLatch(1);
 
 
-    public ApplicationClientDeploymentService(final File path) {
+    public ApplicationClientDeploymentService(final Consumer<ApplicationClientDeploymentService> consumer,
+                                              final File path,
+                                              final Supplier<ModelControllerClientFactory> clientFactorySupplier,
+                                              final Supplier<ExecutorService> executorServiceSupplier) {
+        this.consumer = consumer;
         this.path = path;
+        this.clientFactorySupplier = clientFactorySupplier;
+        this.executorServiceSupplier = executorServiceSupplier;
     }
 
     @Override
-    public synchronized void start(final StartContext context) throws StartException {
-        controllerClient = controllerValue.getValue().createClient(Executors.newSingleThreadExecutor());
-
+    public synchronized void start(final StartContext context) {
         final DeployTask task = new DeployTask();
+        // TODO use executorServiceSupplier
```

Comment:
Any reason not to do this now?

Reply:
@pferraro It was out of scope. I'll file a JIRA re this and your other comment, but I don't want to change the semantics of this for this work, which is aimed at getting rid of deprecated API use so we can drop the API. I'd like to get core 21 Beta1 and 29 Alpha1 done early next week and there's not much time to react if this breaks something.

Reply:
https://issues.redhat.com/browse/WFLY-17952